Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -2173,9 +2173,22 @@ private void UpdateResourceAvailability(PerformanceCollector.MachinePerfInfo per
             }
 
 #if PLATFORM_OSX
-            Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
-            var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
-            resourceAvailable &= !(result && (pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel));
+            if (!resourceAvailable)
+            {
+                Memory.PressureLevel pressureLevel = Memory.PressureLevel.Normal;
+                var result = Memory.GetMemoryPressureLevel(ref pressureLevel) == Dispatch.MACOS_INTEROP_SUCCESS;
+                resourceAvailable = !(result && (pressureLevel > m_configuration.Schedule.MaximumAllowedMemoryPressureLevel));

```

Comment:
This implies that if `result` is false (i.e., we were not able to obtain memory pressure level), we set `resourceAvailable` to `true`.

That looks strange to me.  It would make more sense to me that if we are unable to calculate memory pressure, we just use the previous metric for computing `resourceAvailable` (which in this case already determined that resources are not available)

Reply:
My idea is to have unbounded resource usage here to identify issues because we get jetsam'ed, this way we could end up with a lot of premature cancelation (in the case we can't read memory pressure) and very long running builds. I guess the latter is a better indicator that something is wrong than getting killed by the system.